### ISSUE TITLE
de-client --show-config fails

### ISSUE THREAD

**Issue by DmitryLitvintsev** [HEPCloud/decisionengine_modules#48]
[root@cmsde01 decisionengine]# de-client --show-config
Traceback (most recent call last):
  File "/usr/bin/de-client", line 98, in <module>
    conf = s.show_config()
  File "/usr/lib64/python2.7/xmlrpclib.py", line 1233, in __call__
    return self.__send(self.__name, args)
  File "/usr/lib64/python2.7/xmlrpclib.py", line 1591, in __request
    verbose=self.__verbose
  File "/usr/lib64/python2.7/xmlrpclib.py", line 1273, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/usr/lib64/python2.7/xmlrpclib.py", line 1306, in single_request
    return self.parse_response(response)
  File "/usr/lib64/python2.7/xmlrpclib.py", line 1482, in parse_response
    return u.close()
  File "/usr/lib64/python2.7/xmlrpclib.py", line 794, in close
    raise Fault(**self._stack[0])
xmlrpclib.Fault: <Fault 1: "<type 'exceptions.TypeError'>:dictionary key must be string">

**Comment by DmitryLitvintsev** [comment]
the issue is with dictionary in job_specification.conf that looks like 

```
            'match_expressions': {
#                Bucket 1, match all resources
#                ("Foobar !=2 and VO=='cms' and RequestCpus==1 and (RequestMemory/RequestCpus<=1001.0) and MaxWallTimeMins<=960 and 
RequestMaxInputRate<400","cms_all"): ["GLIDEIN_Supported_VOs == 'CMS' and GLIDEIN_CPUS >= 1 and GLIDEIN_Max_Walltime>=57600" ],
```

The RPC library we use does not support it:

https://docs.python.org/2/library/xmlrpclib.html

Suggest to rewrite dictionary to conform to supported types:

 	dict. Keys must be strings, values may be any conformable type. Objects of user-defined classes can be passed in; only their __dict__ attribute is transmitted.

**Comment by DmitryLitvintsev** [comment]
I believe his suggestion would fix the issue but it is quite hard to actually try it, given the current structure of the job_classication.conf config files.
I understand that xml-rpc does not like to print out dictionaries that have a tuple as a key,
which all the job_classification.conf channels do.
To fix this it would take agreement of the developers on how best to modify the  job classification transform to use a different keying structure.  I leave it to the development crew if they want to do this or not, and on what time scale.

**Comment by DmitryLitvintsev** [comment]
imported from GitLab: https://hepcloud-git.fnal.gov:8443/hepcloud/decisionengine_modules/issues/163

**Comment by StevenCTimm** [comment]
Appears that this issue will be addressed when the framework configuration changes are implemented because it will no longer be possible to have a tuple as a key in the new Jsonnet based configuration system.

**Comment by StevenCTimm** [comment]
This will be resolved when #244 is solved.

**Comment by StevenCTimm** [comment]
Testing CI rpm from master as built July 2.  config files had to be changed to remove any tuples either in keys or in values.  This has been done, and in this CI rpm, the de-client --show-config, --show-de-config,  and --show-channel-config <channel> all work.

This issue can close as soon as the release is cut.

**Comment by StevenCTimm** [comment]
This is fixed in release 1.3.0.  Closing.
